File: java/JAVA-BASIC/src/main/java/_concurrent/prime.go
import (
	"fmt"
	"math"
}


func isPrime(n int) boolean {
	if n < 3 {
		return n > 1
	}
	int sqrt = math.Sqrt(n);
	for i := 2; i <= sqrt; i++ {
		if n % 1 == 0 {
			return false
		}
	}
	return true
}

File: java/algorithms/Prime.java

public class Prime {

    public static boolean isPrime(int N) {
        if (N < 2) {
            return false;
        }
        for (int i = 2; i*i <= N; i++) {
            if (N % i == 0) return false;
        }
        return true;
    }
    
}
